File: javascript/leetcode/122. Best Time to Buy and Sell Stock II.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  if (!prices.length) return 0;
  let max = 0;

  prices.forEach((el, ind) => {
    if (el > prices[ind - 1]) max += el - prices[ind - 1];
  });

  return max;
};

File: c/leetcode/122. Best Time to Buy and Sell Stock II.c


int maxProfit(int *prices, int pricesSize)
{
  if (!pricesSize)
    return 0;
  int max = 0;
  for (int i = 1; i < pricesSize; i++)
  {
    if (prices[i] > prices[i - 1])
      max += prices[i] - prices[i - 1];
  }

  return max;
}